src/audio_pixel.h

```diff
@@ -37,28 +40,64 @@ class audio_pixel_t {
 class audio_pixel_block_t {
 
 public: 
-    // TODO: handle the accessor not being valid
-    audio_pixel_block_t(MediaItem_Take* take, double pix_per_s) 
-        : m_take(take), m_pix_per_s(pix_per_s), 
-          m_accessor(safe_audio_accessor_t(take)) 
-          { /* TODO: */}
+    audio_pixel_block_t(double pix_per_s)
+        :m_pix_per_s(pix_per_s) {}
     ~audio_pixel_block_t() {}
 
-    void update() { /* TODO: see AudioAccessorStateChanged */}
+    void update(std::vector<double>& sample_buffer, int num_channels, int sample_rate) {
+        // constructs an audio pixel block using interleaved sample buffer and num channels
+        // clear the current set of audio pixels 
+        m_channel_pixels.clear();
+
+        // initalize the m_channel_pixels to have empty audio pixel vectors
```

Comment:
awesome! The amount of memory that we need (the number of channels (and number of samples)) in a `MediaItem_Take` is unlikely to change every time we call `update()`. In fact, it probably only changes if the user records more audio (and it may just create a new `MediaItem_Take` in that case, who knows). 

So I think it would be wise to allocate this memory only once, upon creation of the `audio_pixel_block_t`. Of course, if the `AudioAccessor` state changes, we need to check to see if the number of channels and samples has. changed too, and allocate more memory in that case. 

  So here's my recommendation: 
  - abstract away the memory allocation to a separate member function that gets called upon construction (only if the audio accessor is valid, ofc). 
  - in the update loop, check if the accessor state has changed. if it has, check if the number of channels // number of samples has changed. if that has changed too, allocate more memory. have a look at [this](https://en.cppreference.com/w/cpp/container/vector/resize) for resizing. 
  
  Note that this means that the `audio_pixel_block_t` will need to know a little bit more about the world upon initialization, like the number of samples and channels to allocate for, which we should be able to calculate given a sample rate and a pix_per_s resolution